docs/OpenShift GitOps CLI User Guide.md

```diff
@@ -0,0 +1,592 @@
+  GitCommit: c5ea5c4df52943a6fff6c0be181fde5358970304
+  GitTreeState: clean
+  GoVersion: go1.20.10
+  Compiler: gc
+  Platform: linux/amd64
+  ExtraBuildInfo: openshift-gitops-version: 1.11.0, release: 0718122023
+```
+**Note:** The above output is just for reference. The actual details might be different based on the version of OpenShift GitOps argocd CLI client installed.
+
+## Configuring OpenShift GitOps CLI (argocd)
+
+Configure the Red Hat OpenShift GitOps `argocd` CLI to enable tab completion.
+
+### Enabling tab completion
+
+After you install the `argocd` CLI, you can enable tab completion to automatically complete `argocd` commands or suggest options when you press Tab.
+
+#### Prerequisites
+- You must have the `argocd` CLI tool installed.
+- You must have bash-completion installed on your local system.
+
+####  Procedure
+The following procedure enables tab completion for Bash.
+
+1. Save the Bash completion code to a file:
+  ```
+  $ argocd completion bash > argocd_bash_completion
+  ```
+2. Copy the file to /etc/bash_completion.d/:
+  ```
+  $ sudo cp argocd_bash_completion /etc/bash_completion.d/
+  ```
+  Alternatively, you can save the file to a local directory and source it from your `.bashrc` file instead.
+
+Tab completion is enabled when you open a new terminal.
+
+## OpenShift GitOps argocd reference
+
+This section lists the basic `argocd` CLI commands.
+**Note** MicroShift based installation do not host an ArgoCD server and supports only the `core` mode of execution. 
+In the `core` mode (`--core` argument specified), the CLI talks directly to the Kubernetes API server set as per the `KUBECONFIG` environment variable or the default file `$HOME/.kube/config`. There is no need for users to login into the ArgoCD server for executing commands.
+
+### Basic syntax
+
+#### Normal mode
+
+In the normal mode, users have to login to the ArgoCD server component using the login component before executing the commands.
+
+  1. Get the admin password for the ArgoCD server
+      ```
+      ADMIN_PASSWD=$(kubectl get secret argocd-initial-admin-secret -n argocd -o jsonpath='{.data.password}' | base64 -d)
+      ```
+  2. Login to the ArgoCD server using the login command
+      ```
+      argocd login --username admin --password ${ADMIN_PASSWD} <server url>
+      #eg:
+      argocd login --username admin --password ${ADMIN_PASSWD} openshift-gitops.openshift-gitops.apps-crc.testing
+      ```
+  3. Execute the argocd commands
+      ```
+      argocd [command or options] [arguments…​]
+      ```
+
+#### Core mode
+
+In the `core` mode (`--core` argument specified), the CLI talks directly to the Kubernetes API server set as per the `KUBECONFIG` environment variable or the default file `$HOME/.kube/config`. There is no need for users to login into the ArgoCD server for executing commands. The commands would be run as user configured in the kubeconfig file. 
+
+  1. With the default context in kubeconfig file
```

Comment:
@anandf  I guess what you mean is .. for step 1 -- Use this default context in kubeconfig file 

and then step 2 is optional if users want to customize the context in case they are using multiple ArgoCD instances. If so, then they must have to use the step 2 option to set the context to interact with the appropriate ArgoCD instance of their choice. Right?

My only confusion is, do they have to edit this `[context]` parameter? Can we have a quick call tomorrow when you are available to understand this better?

Can we give an example here? I think that will help. WDYT?

Reply:
Yes. Instead of numbered steps, I have put them as `Options` so users can choose one of the available options to run the commands. Yes `[context]` is a placeholder which users have to fill in. I have provided detailed examples for each option.

Reply:
Thanks @anandf for the clarifications and additional inputs.